extension/zpagesextension/config.go

```diff
@@ -7,11 +7,13 @@ import (
 	"errors"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
 	"go.opentelemetry.io/collector/config/confignet"
 )
 
 // Config has the configuration for the extension enabling the zPages extension.
 type Config struct {
+	confighttp.ServerConfig `mapstructure:",squash"`
```

Comment:
You are adding `confighttp.ServerConfig`, which includes the `Endpoint` property. You should remove the `TCPAddr` then, as it is not needed anymore.